test/framework/operator.go

```diff
@@ -120,10 +122,47 @@ func IsKogitoOperatorRunning(namespace string) (bool, error) {
 func WaitForKogitoOperatorRunning(namespace string) error {
 	return WaitForOnOpenshift(namespace, "Kogito operator running", kogitoOperatorTimeoutInMin,
 		func() (bool, error) {
-			return IsKogitoOperatorRunning(namespace)
+			running, err := IsKogitoOperatorRunning(namespace)
+			if err != nil {
+				return false, err
+			}
+
+			// If not running, make sure the image pull secret is present in pod
+			// If not present, delete the pod to allow its reconstruction with correct pull secret
+			// Note that this is specific to Openshift
+			if !running {
+				podList, err := GetPodsWithLabels(namespace, map[string]string{"name": "kogito-operator"})
+				if err != nil {
+					GetLogger(namespace).Errorf("Error while trying to retrieve Kogito Operator pods: %v", err)
+					return false, nil
+				}
+				for _, pod := range podList.Items {
+					if !podHasImagePullSecretPrefix(&pod, kogitoOperatorPullImageSecretPrefix) {
+						// Delete pod as it has been misconfigured (missing pull secret)
+						GetLogger(namespace).Info("Kogito Operator pod does not have the image pull secret needed. Deleting it to renew it.")
+						err := kubernetes.ResourceC(kubeClient).Delete(&pod)
+						if err != nil {
+							GetLogger(namespace).Errorf("Error while trying to delete Kogito Operator pod: %v", err)
+							return false, nil
+						}
+					}
+				}
+			}
+			return running, nil
 		})
 }
 
+func podHasImagePullSecretPrefix(pod *corev1.Pod, imagePullSecretPrefix string) bool {
```

Comment:
This method should be part of the kubernetes script instead?

Reply:
will change that